internal/provider/data_source_http.go

```diff
@@ -163,6 +163,7 @@ func (d *httpDataSource) Read(ctx context.Context, req datasource.ReadRequest, r
 	caCertificate := model.CaCertificate
 
 	tr := &http.Transport{
+		Proxy:           http.ProxyFromEnvironment,
```

Comment:
I'm not sure this is the most complete way to handle behavior differences that were introduced in v3.2.0, as the Proxy is only one of several properties that are set on the DefaultTransport (https://github.com/golang/go/blob/f3c39a83a3076eb560c7f687cbb35eef9b506e7d/src/net/http/transport.go#L38-L54) that are no longer going to be consistent with the previous version behavior.  Ignoring the proxy values was the most immediately noticeable to users, but what about all of the timeouts, `MaxIdleCons`, and `ForceAttemptHTTP2` which will also default to different values than those from older provider versions?

Reply:
I would also recommend using `http.DefaultTransport` to prevent other potential behavior changes. There could also be consideration for using https://pkg.go.dev/github.com/hashicorp/go-cleanhttp similar to other HashiCorp maintained Go modules to prevent potential issues with multiple host connections across data sources or other regressions involving a mutated `http.DefaultTransport`.